app/services/alerts/collate_benchmark_data.rb

```diff
@@ -3,24 +3,28 @@ class CollateBenchmarkData
     def perform(schools = School.process_data)
       benchmarks = {}
 
-      schools.each do |school|
-        get_benchmarks_for(school, benchmarks)
-      end
+      latest_run = BenchmarkResultGenerationRun.order(created_at: :desc).first
+      latest_school_runs = latest_run.benchmark_result_school_generation_runs.where(school: schools)
+      get_benchmarks_for_latest_run(latest_school_runs, benchmarks)
 
       benchmarks
     end
 
     private
 
-    def get_benchmarks_for(school, benchmarks)
-      school.latest_benchmark_results.each do |benchmark_result|
-        unless benchmarks.key?(benchmark_result.asof)
-          benchmarks[benchmark_result.asof] = { school.id => {} }
-        end
-        unless benchmarks[benchmark_result.asof].key?(school.id)
-          benchmarks[benchmark_result.asof][school.id] = {}
+    def get_benchmarks_for_latest_run(latest_school_runs, benchmarks)
+      latest_school_runs.each do |benchmark_result_school_generation_run|
+        school = benchmark_result_school_generation_run.school
```

Comment:
If we're just using the ID, can we not just get the `school_id` from the benchmark_result_school_generation_run and save the school load?

Reply:
YES!